src/lava_callback.py

```diff
@@ -211,6 +210,23 @@ def async_job_submit(api_helper, node_id, job_callback):
     job_result = job_callback.get_job_status()
     device_id = job_callback.get_device_id()
     storage_config_name = job_callback.get_meta('storage_config_name')
+    input_file_types = [
+      'nfsrootfs',
+      'rootfs',
+      'ramdisk',
+      'initrd',
+      'ndbroot',
+      'persistent_nfs'
+    ]
+    job_actions = job_callback.get_job_definition('actions')
+    if job_actions and len(job_actions) > 0:
+        deploy = job_actions[0].get('deploy')
+        if deploy:
+            for input_file in input_file_types:
+                if deploy.get('images') and deploy['images'].get(input_file):
+                    url = deploy['images'][input_file].get('url')
```

Comment:
To reduce a bit of nestedness

```suggestion
        if deploy_images := job_actions[0].get('deploy', {}).get('images'):
            for input_file in input_file_types:
                if url := deploy_images.get(input_file, {}).get('url'):
                    job_node['artifacts']["input_"+input_file] = url
```

I'm assuming we are using python>=3.8 and relying in the walrus operator, but it's not required